ui-performance.md

```diff
@@ -0,0 +1,470 @@
+to `saveLayer` and the presence of cached images.
+
+<aside class="alert alert-info" markdown="1">
+**Note:**
+You are probably familiar with the Flutter Gallery example app. To use the performance overlay
+with Flutter Gallery, download, and run the app in profile mode. The program is written
+so that the app menu allows you to dynamically toggle the overlay, as well as enable checks
+for calls to saveLayer and the presence of cached images.
+</aside>
+
+### Identifying problems in the engine graph
+
+If the performance overlay shows red in the engine graph, start by profiling the engine (Dart VM),
+even if the rasterizer graph also shows red. To do this use
+[Observatory](https://dart-lang.github.io/observatory/), Dart’s profiling tool.
+
+#### Displaying Observatory
+
+Observatory provides a statement-level single-stepping debugger for Dart and Flutter applications.
+ It also supports profiling, examining the heap, and reporting on code coverage. For this doc,
+we are using Observatory’s timeline view to capture a snapshot of the timeline.
+[TBD: more text about the timeline view]
+
+Display Observatory in a browser as follows:
+
+<ol markdown="1">
+<li markdown="1">
+When launching a Flutter app from the command line, you’ll see text similar to the following:
+
+{% prettify sh %}
+An Observatory debugger and profiler on <device> is available at: http://127.0.0.1:8100/
+{% endprettify %}
+</li>
+<li markdown="1">
+In Android Studio or IntelliJ,
+the Observatory icon (<img src="images/observatory-icon.png" alt="stopwatch icon">)
+appears in the row of icons next to the Console at the bottom of the screen.
+It also appears in the Flutter Inspector, next to the gear icon.
+Click the stopwatch icon to open Observatory’s timeline view in a browser.
+[PENDING: Provide a screenshot when the UI is stable.]
+</li>
+<li markdown="1">
+In VSCode, bring up the command palette and enter “observatory”.
+Select **Flutter: Open Observatory Timeline** from the list that pops up.
+If this command isn’t available, make sure that the app is running.
+[PENDING: Provide a screenshot when the UI is stable.]
+</li>
+</ol>
+
+#### Using Observatory's timeline
+
+Observatory is a web application that is hosted by the VM. It provides a suite of
+program inspection and analysis tools, but we are primarily using the timeline to
+grab and view slices of the stack to determine where your Flutter app is spending
+its time.
+
+<ol markdown="1">
+
+<li markdown="1"> When bringing up Observatory from an IDE,
+    it takes you directly to the timeline view.
+    If you are looking at Observatory’s front page,
+    click the “view <ul>timeline</ul>” link.
+
+    Initially, timeline page contains no timeline data. In the **Recorded Streams Profile**,
+    select the **Flutter Developer profile** radio button. This simultaneously selects the **Dart**,
+    **Embedder**, and **GC** streams.
+</li>
+
+<li markdown="1"> Run the app and get it to the state that you want to examine.
```

Comment:
I'd use a different word than 'run' since the app is already running (that's how we got the observatory URL).